Include ansible playbook for test-disk-failure.

nsathyaseelan opened the pull request:
Signed-off-by: nsathyaseelan <sathyaseelan47@gmail.com>

<!--  Thanks for sending a pull request!  Here are some tips for you -->

**What this PR does / why we need it**:
- Includes an ansible playbook test disk failure Condition and verify the behavior.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

Playbook Output:
```
root@sathya-virtual-machine:/home/sathya/openebs/e2e/ansible# ansible-playbook playbooks/resiliency/test-disk-failure/test-disk-failure.yaml -e "uuid=f536e2da-fcf5-410d-8955-a096d1048b4f" -vv
ansible-playbook 2.5.2
  config file = /home/sathya/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/sathya/openebs/e2e/ansible/ansible.cfg as config file

PLAYBOOK: test-disk-failure.yaml ***************************************************************************************************************************************
1 plays in playbooks/resiliency/test-disk-failure/test-disk-failure.yaml

PLAY [localhost] *******************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:5
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] *********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:13
changed: [localhost -> None] => {"changed": true, "cmd": "echo $HOME", "delta": "0:00:00.004660", "end": "2018-05-18 17:56:49.249117", "rc": 0, "start": "2018-05-18 17:56:49.244457", "stderr": "", "stderr_lines": [], "stdout": "/root", "stdout_lines": ["/root"]}

TASK [Get the disk name by using UUID] *********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:21
changed: [localhost -> None] => {"changed": true, "cmd": "blkid | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $1}' | cut -d \"/\" -f3 | cut -d \":\" -f1", "delta": "0:00:00.131523", "end": "2018-05-18 17:56:51.051556", "rc": 0, "start": "2018-05-18 17:56:50.920033", "stderr": "", "stderr_lines": [], "stdout": "sdc", "stdout_lines": ["sdc"]}

TASK [Get the Mount point of the disk using UUID] **********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:29
changed: [localhost -> None] => {"changed": true, "cmd": "lsblk -f | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $4}'", "delta": "0:00:00.018219", "end": "2018-05-18 17:56:51.446816", "rc": 0, "start": "2018-05-18 17:56:51.428597", "stderr": "", "stderr_lines": [], "stdout": "/mnt/test", "stdout_lines": ["/mnt/test"]}

TASK [Check status of maya-api server] *********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:37
included: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check maya-api pod status] ***************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n default | grep maya-api", "delta": "0:00:00.230462", "end": "2018-05-18 17:56:52.122657", "rc": 0, "start": "2018-05-18 17:56:51.892195", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "maya-apiserver-1255925684-ksx7v        1/1       Running   0          1h", "stdout_lines": ["maya-apiserver-1255925684-ksx7v        1/1       Running   0          1h"]}

TASK [Get the storage pool creation YAML file] *************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:43
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "1539e9d92af1b4196f65e625288e11f541b9a8b7", "dest": "/root/storage-pool-path_creation.yml", "gid": 0, "group": "root", "md5sum": "9dff24343f20b869a6a38d0f493e9505", "mode": "0644", "msg": "OK (128 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 128, "src": "/tmp/tmpB5YvGB", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/openebs/master/e2e/ansible/playbooks/hyperconverged/test-k8s-storage-pool/storage-pool-path_creation.yml"}

TASK [Replace storage pool name with test parameters] ******************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:50
included: /root/openebs/e2e/ansible/playbooks/utils/regex_task.yml for localhost

TASK [Replace volume-claim name with test parameters] ******************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/regex_task.yml:2
changed: [localhost -> None] => (item=[u'default', u'disk']) => {"changed": true, "item": ["default", "disk"], "msg": "1 replacements made"}
changed: [localhost -> None] => (item=[u'/mnt/openebs', u'/mnt/test']) => {"changed": true, "item": ["/mnt/openebs", "/mnt/test"], "msg": "1 replacements made"}

TASK [create the storage pool path] ************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:57
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f storage-pool-path_creation.yml", "delta": "0:00:00.213817", "end": "2018-05-18 17:56:56.103853", "rc": 0, "start": "2018-05-18 17:56:55.890036", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "storagepool \"disk\" created", "stdout_lines": ["storagepool \"disk\" created"]}

TASK [Obtaining storage classes yaml] **********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:64
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get sc openebs-percona -o yaml > \"storage-class.yaml\"", "delta": "0:00:00.243599", "end": "2018-05-18 17:56:56.728619", "rc": 0, "start": "2018-05-18 17:56:56.485020", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Replace storage class file with test parameters] *****************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:70
changed: [localhost -> None] => {"changed": true, "msg": "1 replacements made"}

TASK [Delete the existing storage class and create new one] ************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:77
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl delete sc openebs-percona; kubectl apply -f \"storage-class.yaml\"", "delta": "0:00:00.499657", "end": "2018-05-18 17:56:57.958787", "rc": 0, "start": "2018-05-18 17:56:57.459130", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-percona\" deleted\nstorageclass \"openebs-percona\" created", "stdout_lines": ["storageclass \"openebs-percona\" deleted", "storageclass \"openebs-percona\" created"]}

TASK [Obtain the sql test script file] *********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:87
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "1808253d88759a699ea9738916d1febe04547ca6", "dest": "/root/sql-test.sh", "gid": 0, "group": "root", "md5sum": "d235f8179f540ebae853378070247fd5", "mode": "0644", "msg": "OK (399 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 399, "src": "/tmp/tmpZK3Cvw", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/sql-test.sh"}

TASK [Create the configmap test] ***************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:93
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl create configmap sqltest --from-file=sql-test.sh", "delta": "0:00:00.216208", "end": "2018-05-18 17:57:01.405727", "rc": 0, "start": "2018-05-18 17:57:01.189519", "stderr": "", "stderr_lines": [], "stdout": "configmap \"sqltest\" created", "stdout_lines": ["configmap \"sqltest\" created"]}

TASK [Get the percona liveliness application yaml] *********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:99
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "66bda981b5e740c1715ec7aefc89000f0797c403", "dest": "/root/percona.yaml", "gid": 0, "group": "root", "md5sum": "d7b60974e8970cc7ceb7025ac028290f", "mode": "0644", "msg": "OK (1593 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 1593, "src": "/tmp/tmpwEidJT", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/percona.yaml"}

TASK [Deploying percona application] ***********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:105
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl apply -f percona.yaml", "delta": "0:00:00.411823", "end": "2018-05-18 17:57:04.841167", "rc": 0, "start": "2018-05-18 17:57:04.429344", "stderr": "", "stderr_lines": [], "stdout": "deployment \"percona\" created\npersistentvolumeclaim \"demo-vol1-claim\" created\nservice \"percona-mysql\" created", "stdout_lines": ["deployment \"percona\" created", "persistentvolumeclaim \"demo-vol1-claim\" created", "service \"percona-mysql\" created"]}

TASK [Check whether percona application is running] ********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:112
included: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check percona pod status] ****************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
FAILED - RETRYING: Check percona pod status (15 retries left).
FAILED - RETRYING: Check percona pod status (14 retries left).
FAILED - RETRYING: Check percona pod status (13 retries left).
FAILED - RETRYING: Check percona pod status (12 retries left).
FAILED - RETRYING: Check percona pod status (11 retries left).
FAILED - RETRYING: Check percona pod status (10 retries left).
changed: [localhost -> None] => {"attempts": 7, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n default | grep percona", "delta": "0:00:00.280911", "end": "2018-05-18 18:00:09.371108", "rc": 0, "start": "2018-05-18 18:00:09.090197", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "percona-2897667160-tl5vr                                         1/1       Running   0          3m", "stdout_lines": ["percona-2897667160-tl5vr                                         1/1       Running   0          3m"]}

TASK [Check Replica  pod status] ***************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:118
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods --all-namespaces | grep pvc", "delta": "0:00:00.281210", "end": "2018-05-18 18:00:10.064563", "rc": 0, "start": "2018-05-18 18:00:09.783353", "stderr": "", "stderr_lines": [], "stdout": "default                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-ctrl-2309976944-rbm8v   2/2       Running            0          3m\ndefault                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bn27f    0/1       Running            0          3m\ndefault                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bvz5x    1/1       Running            0          3m\ndefault                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-w2d7p    1/1       Running            0          3m", "stdout_lines": ["default                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-ctrl-2309976944-rbm8v   2/2       Running            0          3m", "default                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bn27f    0/1       Pending            0          3m", "default                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bvz5x    1/1       Running            0          3m", "default                             pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-w2d7p    1/1       Running            0          3m"]}

TASK [Failing the additional disk] *************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:128
changed: [localhost -> None] => {"changed": true, "cmd": "echo 1 > /sys/block/sdc/device/delete", "delta": "0:00:00.003303", "end": "2018-05-18 18:00:11.876624", "rc": 0, "start": "2018-05-18 18:00:11.873321", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [List the available scsi host] ************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:135
changed: [localhost -> None] => {"changed": true, "cmd": "ls /sys/class/scsi_host/", "delta": "0:00:00.004208", "end": "2018-05-18 18:00:12.216457", "rc": 0, "start": "2018-05-18 18:00:12.212249", "stderr": "", "stderr_lines": [], "stdout": "host0\nhost1\nhost2\nhost5", "stdout_lines": ["host0", "host1", "host2", "host5"]}

TASK [Rescan the disk] *************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:143
changed: [localhost -> None] => (item=host0) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host0/scan", "delta": "0:00:00.005874", "end": "2018-05-18 18:00:12.617454", "item": "host0", "rc": 0, "start": "2018-05-18 18:00:12.611580", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host1) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host1/scan", "delta": "0:00:00.165437", "end": "2018-05-18 18:00:13.081132", "item": "host1", "rc": 0, "start": "2018-05-18 18:00:12.915695", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host2) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host2/scan", "delta": "0:00:00.163764", "end": "2018-05-18 18:00:13.533591", "item": "host2", "rc": 0, "start": "2018-05-18 18:00:13.369827", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host5) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host5/scan", "delta": "0:00:00.004419", "end": "2018-05-18 18:00:13.828944", "item": "host5", "rc": 0, "start": "2018-05-18 18:00:13.824525", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Checking the  Mount point using UUID after failing the disk] *****************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:151
changed: [localhost -> None] => {"changed": true, "cmd": "lsblk -f | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $4}'", "delta": "0:00:00.016425", "end": "2018-05-18 18:00:14.223801", "rc": 0, "start": "2018-05-18 18:00:14.207376", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Get the disk name by using UUID] *********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:159
changed: [localhost -> None] => {"changed": true, "cmd": "blkid | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $1}' | cut -d \"/\" -f3 | cut -d \":\" -f1", "delta": "0:00:00.223838", "end": "2018-05-18 18:00:14.829252", "rc": 0, "start": "2018-05-18 18:00:14.605414", "stderr": "", "stderr_lines": [], "stdout": "sde", "stdout_lines": ["sde"]}

TASK [Remount the disk] ************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:167
 [WARNING]: Consider using the mount module rather than running mount.  If you need to use command because mount is insufficient you can add warn=False to this command
task or set command_warnings=False in ansible.cfg to get rid of this message.

changed: [localhost -> None] => {"changed": true, "cmd": "mount /dev/sde /mnt/test", "delta": "0:00:00.065756", "end": "2018-05-18 18:00:15.251843", "rc": 0, "start": "2018-05-18 18:00:15.186087", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Check Replica  pod status] ***************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yaml:174
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods | grep pvc", "delta": "0:00:00.272885", "end": "2018-05-18 18:00:16.005569", "rc": 0, "start": "2018-05-18 18:00:15.732684", "stderr": "", "stderr_lines": [], "stdout": "pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-ctrl-2309976944-rbm8v   2/2       Running   0          3m\npvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bn27f    0/1       Running   0          3m\npvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bvz5x    1/1       Running   0          3m\npvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-w2d7p    1/1       Running   0          3m", "stdout_lines": ["pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-ctrl-2309976944-rbm8v   2/2       Running   0          3m", "pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bn27f    0/1       Pending   0          3m", "pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-bvz5x    1/1       Running   0          3m", "pvc-c5f6abff-5a96-11e8-bdfd-0050569b107a-rep-1576092488-w2d7p    1/1       Running   0          3m"]}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************
localhost                  : ok=27   changed=23   unreachable=0    failed=0

```

nsathyaseelan commented:
@dargasudarshan @yudaykiran @ksatchit please review this PR

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,179 @@
+  tasks:
+   - block:
+
+       - name: Get $HOME of K8s master for kubernetes user
```
And also include task to initiate stern in the same file.

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,179 @@
+           dest: "{{ result_kube_home.stdout }}"
+         delegate_to: "{{ groups['kubernetes-kubemasters'].0 }}"
+
+       - name: Create the configmap test
```
Deploy test application on its own namespace and the namespace name should be test specific and unique.

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,179 @@
+         args:
+           executable: /bin/bash
+         register: result_status
+         delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
```
Include cleanup task in the playbook and set flag for cleanup status.

dargasudarshan commented:
Lint the playbook and provide the output here.

nsathyaseelan commented:
@dargasudarshan added review comments

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,209 @@
+# test-disk-failure.yaml
+
```
Add your name as author

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,209 @@
+
+       - include: test-disk-failure-prereq.yml
+
+       - name: Check status of maya-api server
```
Add task numbers

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,209 @@
+         vars:
+           app: pvc
+           ns: {{ namespace }}
+
```
Add test case pass fail flags

ksatchit commented:
@nsathyaseelan , can you please provide more details on the PR wrt approach taken to verify disk failure scenarios ?

ksatchit commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+           app: maya-api
+
+       - name: 3) Get the disk name by using UUID
+         shell: blkid | grep {{ uuid }} | awk '{print $1}' | cut -d "/" -f3 | cut -d ":" -f1
```
Can we use the shell module's space/line separator ?

ksatchit commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           app: pvc
+           ns: {{ namespace }}
```
Do we have a check to confirm that the pods are running on the specified storage pool path ?

ksatchit commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+           executable: /bin/bash
+         become: true
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
```
Can we have two more tasks here : 

- Check to verify that the disk is indeed removed on the above step
- Impact on the replica pod scheduled on the node

ksatchit commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+         delegate_to: "{{ groups['kubernetes-kubeminions'].0 }}"
+
+       - name: 11b) Get the disk name using UUID
+         shell: blkid | grep {{ uuid }} | awk '{print $1}' | cut -d "/" -f3 | cut -d ":" -f1
```
Use shell module's space/line separator

ksatchit commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+         include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/deploy_check.yml"
+         vars:
+           app: pvc
+           ns: {{ namespace }}
```
Once the replica pod is running, there has to be a check to see that it has been added back in RW mode, i.e., if dirty data sync has occurred.

dargasudarshan commented on e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:
```diff
@@ -0,0 +1,221 @@
+         include_tasks: "{{ ansible_env.HOME }}/{{utils_path}}/deploy_check.yml"
+         vars:
+           ns: openebs
+           app: maya-api
```
Since we are updating the helm to use stable charts, maya-api server is deprecated and now it will be openeb-api server. Can you make the changes accordingly?

nsathyaseelan commented:
output for the Playbook after Modification.
```
ansible-playbook 2.5.2
  config file = /home/sathya/openebs/e2e/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/dist-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.12 (default, Dec  4 2017, 14:50:18) [GCC 5.4.0 20160609]
Using /home/sathya/openebs/e2e/ansible/ansible.cfg as config file
statically imported: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-cleanup.yml
statically imported: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-prereq.yml

PLAYBOOK: test-disk-failure.yml ****************************************************************************************************************************************
1 plays in playbooks/resiliency/test-disk-failure/test-disk-failure.yml

PLAY [localhost] *******************************************************************************************************************************************************

TASK [Gathering Facts] *************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:20
ok: [localhost]
META: ran handlers

TASK [Get $HOME of K8s master for kubernetes user] *********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-prereq.yml:2
changed: [localhost -> None] => {"changed": true, "cmd": "echo $HOME", "delta": "0:00:00.004094", "end": "2018-05-24 18:28:15.762027", "rc": 0, "start": "2018-05-24 18:28:15.757933", "stderr": "", "stderr_lines": [], "stdout": "/root", "stdout_lines": ["/root"]}

TASK [2) Check status of maya-api server] ******************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:31
included: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check maya-api pod status] ***************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n default | grep maya-api", "delta": "0:00:00.271298", "end": "2018-05-24 18:28:16.458565", "rc": 0, "start": "2018-05-24 18:28:16.187267", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "maya-apiserver-1255925684-9bpcz        1/1       Running   0          2d", "stdout_lines": ["maya-apiserver-1255925684-9bpcz        1/1       Running   0          2d"]}

TASK [3) Get the disk name by using UUID] ******************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:37
changed: [localhost -> None] => {"changed": true, "cmd": "blkid | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $1}' | cut -d \"/\" -f3 | cut -d \":\" -f1", "delta": "0:00:00.116378", "end": "2018-05-24 18:28:18.297478", "rc": 0, "start": "2018-05-24 18:28:18.181100", "stderr": "", "stderr_lines": [], "stdout": "sdag", "stdout_lines": ["sdag"]}

TASK [3a) Get the Mount point of the disk using UUID] ******************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:47
changed: [localhost -> None] => {"changed": true, "cmd": "lsblk -f | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $4}'", "delta": "0:00:00.023440", "end": "2018-05-24 18:28:18.736571", "rc": 0, "start": "2018-05-24 18:28:18.713131", "stderr": "", "stderr_lines": [], "stdout": "/mnt/test", "stdout_lines": ["/mnt/test"]}

TASK [4) Obtaining the storage pool creation yaml file] ****************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:55
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "1539e9d92af1b4196f65e625288e11f541b9a8b7", "dest": "/root/storage-pool-path_creation.yml", "gid": 0, "group": "root", "md5sum": "9dff24343f20b869a6a38d0f493e9505", "mode": "0644", "msg": "OK (128 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 128, "src": "/tmp/tmpmicYOd", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/openebs/master/e2e/ansible/playbooks/hyperconverged/test-k8s-storage-pool/storage-pool-path_creation.yml"}

TASK [4a) Replace storage pool name with test parameters] **************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:62
included: /root/openebs/e2e/ansible/playbooks/utils/regex_task.yml for localhost

TASK [Replace volume-claim name with test parameters] ******************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/regex_task.yml:2
changed: [localhost -> None] => (item=[u'default', u'disk']) => {"changed": true, "item": ["default", "disk"], "msg": "1 replacements made"}
changed: [localhost -> None] => (item=[u'/mnt/openebs', u'/mnt/test']) => {"changed": true, "item": ["/mnt/openebs", "/mnt/test"], "msg": "1 replacements made"}

TASK [5)Create the storage pool path] **********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:75
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f storage-pool-path_creation.yml -n default", "delta": "0:00:00.265118", "end": "2018-05-24 18:28:22.962930", "rc": 0, "start": "2018-05-24 18:28:22.697812", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "storagepool \"disk\" created", "stdout_lines": ["storagepool \"disk\" created"]}

TASK [5a) Obtaining storage classes yaml] ******************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:82
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get sc openebs-percona -o yaml > \"storage-class.yaml\"", "delta": "0:00:00.210054", "end": "2018-05-24 18:28:23.519865", "rc": 0, "start": "2018-05-24 18:28:23.309811", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [5b) Replace storage class file with test parameters] *************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:88
ok: [localhost -> None] => {"changed": false, "msg": ""}

TASK [5c) Delete the existing storage class and create new one] ********************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:95
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "kubectl delete sc openebs-percona; kubectl apply -f \"storage-class.yaml\"", "delta": "0:00:00.433338", "end": "2018-05-24 18:28:24.628242", "rc": 0, "start": "2018-05-24 18:28:24.194904", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-percona\" deleted\nstorageclass \"openebs-percona\" created", "stdout_lines": ["storageclass \"openebs-percona\" deleted", "storageclass \"openebs-percona\" created"]}

TASK [6) Obtain the sql test script file] ******************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:105
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "1808253d88759a699ea9738916d1febe04547ca6", "dest": "/root/sql-test.sh", "gid": 0, "group": "root", "md5sum": "d235f8179f540ebae853378070247fd5", "mode": "0644", "msg": "OK (399 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 399, "src": "/tmp/tmpUKzjdE", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/sql-test.sh"}

TASK [6a) Create the configmap test] ***********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:111
 [WARNING]: As of Ansible 2.4, the parameter 'executable' is no longer supported with the 'command' module. Not using '/bin/bash'.

changed: [localhost -> None] => {"changed": true, "cmd": ["kubectl", "create", "configmap", "sqltest", "--from-file=sql-test.sh", "-n", "default"], "delta": "0:00:00.213485", "end": "2018-05-24 18:28:28.643439", "rc": 0, "start": "2018-05-24 18:28:28.429954", "stderr": "", "stderr_lines": [], "stdout": "configmap \"sqltest\" created", "stdout_lines": ["configmap \"sqltest\" created"]}

TASK [7) Obtaining the percona liveliness application yaml] ************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:117
changed: [localhost -> None] => {"changed": true, "checksum_dest": null, "checksum_src": "66bda981b5e740c1715ec7aefc89000f0797c403", "dest": "/root/percona.yaml", "gid": 0, "group": "root", "md5sum": "d7b60974e8970cc7ceb7025ac028290f", "mode": "0644", "msg": "OK (1593 bytes)", "owner": "root", "secontext": "system_u:object_r:admin_home_t:s0", "size": 1593, "src": "/tmp/tmpfM9RSB", "state": "file", "status_code": 200, "uid": 0, "url": "https://raw.githubusercontent.com/openebs/elves/master/e2e/percona-liveness/percona.yaml"}

TASK [7a) Create percona deployment with OpenEBS storage] **************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:123
included: /root/openebs/e2e/ansible/playbooks/utils/deploy_task.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/deploy_task.yml:2
changed: [localhost -> None] => (item=percona.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl apply -f percona.yaml -n default", "delta": "0:00:00.389749", "end": "2018-05-24 18:28:31.980962", "item": "percona.yaml", "rc": 0, "start": "2018-05-24 18:28:31.591213", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "deployment \"percona\" created\npersistentvolumeclaim \"demo-vol1-claim\" created\nservice \"percona-mysql\" created", "stdout_lines": ["deployment \"percona\" created", "persistentvolumeclaim \"demo-vol1-claim\" created", "service \"percona-mysql\" created"]}

TASK [7c) Check whether percona application is running] ****************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:129
included: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml for localhost

TASK [Check percona pod status] ****************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/deploy_check.yml:2
FAILED - RETRYING: Check percona pod status (15 retries left).
FAILED - RETRYING: Check percona pod status (14 retries left).
FAILED - RETRYING: Check percona pod status (13 retries left).
FAILED - RETRYING: Check percona pod status (12 retries left).
FAILED - RETRYING: Check percona pod status (11 retries left).
FAILED - RETRYING: Check percona pod status (10 retries left).
FAILED - RETRYING: Check percona pod status (9 retries left).
changed: [localhost -> None] => {"attempts": 8, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n default | grep percona", "delta": "0:00:00.263415", "end": "2018-05-24 18:32:08.395721", "rc": 0, "start": "2018-05-24 18:32:08.132306", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "percona-2897667160-zp7p5                                        1/1       Running   0          3m", "stdout_lines": ["percona-2897667160-zp7p5                                        1/1       Running   0          3m"]}

TASK [8) Obtaining the replica pods running on specified node] *********************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:141
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -o wide -n default | grep newos2 | grep rep | awk '{ print $1 }'", "delta": "0:00:00.241813", "end": "2018-05-24 18:32:09.054184", "rc": 0, "start": "2018-05-24 18:32:08.812371", "stderr": "", "stderr_lines": [], "stdout": "pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf", "stdout_lines": ["pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf"]}

TASK [8a) Check Replica pods running on specified storage pool path] ***************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:151
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl describe pod -n default \"pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf\" | grep \"/mnt/test\" | awk '{print $2}'", "delta": "0:00:00.258670", "end": "2018-05-24 18:32:09.752624", "failed_when_result": false, "rc": 0, "start": "2018-05-24 18:32:09.493954", "stderr": "", "stderr_lines": [], "stdout": "/mnt/test/pvc-294114bb-5f52-11e8-8319-0050569b107a", "stdout_lines": ["/mnt/test/pvc-294114bb-5f52-11e8-8319-0050569b107a"]}

TASK [9) Failing the additional disk] **********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:162
changed: [localhost -> None] => {"changed": true, "cmd": "echo 1 > /sys/block/sdag/device/delete", "delta": "0:00:00.003774", "end": "2018-05-24 18:32:12.174568", "rc": 0, "start": "2018-05-24 18:32:12.170794", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [9a) Check the Disk is failing] ***********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:169
changed: [localhost -> None] => {"changed": true, "cmd": "lsblk | grep sdag", "delta": "0:00:00.021330", "end": "2018-05-24 18:32:12.555537", "failed_when_result": false, "msg": "non-zero return code", "rc": 1, "start": "2018-05-24 18:32:12.534207", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [10) Check the replica pod on the specified node] *****************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:178
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -o wide -n default | grep newos2 | grep rep | awk '{ print $1 }'", "delta": "0:00:00.245965", "end": "2018-05-24 18:32:13.219168", "rc": 0, "start": "2018-05-24 18:32:12.973203", "stderr": "", "stderr_lines": [], "stdout": "pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf", "stdout_lines": ["pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf"]}

TASK [10) Delete the replica pod on the specified node] ****************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:186
changed: [localhost -> None] => {"changed": true, "cmd": ["kubectl", "delete", "pod", "pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf"], "delta": "0:00:00.226927", "end": "2018-05-24 18:32:13.854016", "rc": 0, "start": "2018-05-24 18:32:13.627089", "stderr": "", "stderr_lines": [], "stdout": "pod \"pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf\" deleted", "stdout_lines": ["pod \"pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-6g4zf\" deleted"]}

TASK [Wait for 120s to ensure liveness check starts] *******************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:193
ok: [localhost] => {"changed": false, "elapsed": 120, "path": null, "port": null, "search_regex": null, "state": "started"}

TASK [10) Check the replica pod status scheduled on the specified node] ************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:197
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -o wide -n default | grep newos2 | grep rep | awk '{ print $3 }'", "delta": "0:00:00.275171", "end": "2018-05-24 18:34:16.197191", "failed_when_result": false, "rc": 0, "start": "2018-05-24 18:34:15.922020", "stderr": "", "stderr_lines": [], "stdout": "CrashLoopBackOff", "stdout_lines": ["CrashLoopBackOff"]}

TASK [10a) Check the percona application pod status] *******************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:206
changed: [localhost -> None] => {"changed": true, "cmd": "kubectl get pods -o wide -n default | grep percona | awk '{ print $3 }'", "delta": "0:00:00.227968", "end": "2018-05-24 18:34:16.825094", "rc": 0, "start": "2018-05-24 18:34:16.597126", "stderr": "", "stderr_lines": [], "stdout": "Running", "stdout_lines": ["Running"]}

TASK [11) List the available scsi host] ********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:215
changed: [localhost -> None] => {"changed": true, "cmd": ["ls", "/sys/class/scsi_host/"], "delta": "0:00:00.004113", "end": "2018-05-24 18:34:18.601099", "rc": 0, "start": "2018-05-24 18:34:18.596986", "stderr": "", "stderr_lines": [], "stdout": "host0\nhost1\nhost2\nhost28\nhost31\nhost41", "stdout_lines": ["host0", "host1", "host2", "host28", "host31", "host41"]}

TASK [11a) Rescan the disk] ********************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:221
changed: [localhost -> None] => (item=host0) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host0/scan", "delta": "0:00:00.029070", "end": "2018-05-24 18:34:19.092874", "item": "host0", "rc": 0, "start": "2018-05-24 18:34:19.063804", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host1) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host1/scan", "delta": "0:00:00.167579", "end": "2018-05-24 18:34:19.630266", "item": "host1", "rc": 0, "start": "2018-05-24 18:34:19.462687", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host2) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host2/scan", "delta": "0:00:00.166749", "end": "2018-05-24 18:34:20.158103", "item": "host2", "rc": 0, "start": "2018-05-24 18:34:19.991354", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host28) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host28/scan", "delta": "0:00:00.005003", "end": "2018-05-24 18:34:20.535206", "item": "host28", "rc": 0, "start": "2018-05-24 18:34:20.530203", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host31) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host31/scan", "delta": "0:00:00.004022", "end": "2018-05-24 18:34:20.884998", "item": "host31", "rc": 0, "start": "2018-05-24 18:34:20.880976", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}
changed: [localhost -> None] => (item=host41) => {"changed": true, "cmd": "echo \"- - -\" > /sys/class/scsi_host/host41/scan", "delta": "0:00:00.004278", "end": "2018-05-24 18:34:21.241879", "item": "host41", "rc": 0, "start": "2018-05-24 18:34:21.237601", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [12) Get the disk name by using UUID] *****************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:229
changed: [localhost -> None] => {"changed": true, "cmd": "blkid | grep f536e2da-fcf5-410d-8955-a096d1048b4f | awk '{print $1}' | cut -d \"/\" -f3 | cut -d \":\" -f1", "delta": "0:00:00.215868", "end": "2018-05-24 18:34:21.884420", "rc": 0, "start": "2018-05-24 18:34:21.668552", "stderr": "", "stderr_lines": [], "stdout": "sdah", "stdout_lines": ["sdah"]}

TASK [12a) Remount the disk] *******************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:239
 [WARNING]: Consider using the mount module rather than running mount.  If you need to use command because mount is insufficient you can add warn=False to this command
task or set command_warnings=False in ansible.cfg to get rid of this message.

changed: [localhost -> None] => {"changed": true, "cmd": "mount /dev/sdah /mnt/test", "delta": "0:00:00.072205", "end": "2018-05-24 18:34:22.416510", "rc": 0, "start": "2018-05-24 18:34:22.344305", "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [Get replica pod name] ********************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:254
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get pods -o wide | grep rep | grep newos2 | awk {'print $1'}", "delta": "0:00:00.269512", "end": "2018-05-24 18:34:23.083838", "rc": 0, "start": "2018-05-24 18:34:22.814326", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-vtv4p", "stdout_lines": ["pvc-294114bb-5f52-11e8-8319-0050569b107a-rep-3431871836-vtv4p"]}

TASK [14) Get Controller SVC IP] ***************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:261
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl get svc -n default | grep ctrl | awk {'print $2'}", "delta": "0:00:00.250667", "end": "2018-05-24 18:34:23.696439", "rc": 0, "start": "2018-05-24 18:34:23.445772", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "172.30.37.199", "stdout_lines": ["172.30.37.199"]}

TASK [Check if replica is rebuilded] ***********************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:271
FAILED - RETRYING: Check if replica is rebuilded (10 retries left).
FAILED - RETRYING: Check if replica is rebuilded (9 retries left).
 [WARNING]: Consider using the get_url or uri module rather than running curl.  If you need to use command because get_url or uri is insufficient you can add
warn=False to this command task or set command_warnings=False in ansible.cfg to get rid of this message.

changed: [localhost -> None] => {"attempts": 3, "changed": true, "cmd": "curl GET http://172.30.37.199:9501/v1/replicas | grep createTypes | jq -r '.data[].mode' | grep 'RW'", "delta": "0:00:00.017791", "end": "2018-05-24 18:36:29.072128", "rc": 0, "start": "2018-05-24 18:36:29.054337", "stderr": "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n                                 Dload  Upload   Total   Spent    Left  Speed\n\r  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: GET; No address associated with hostname\n\r100  1041  100  1041    0     0   708k      0 --:--:-- --:--:-- --:--:--  708k", "stderr_lines": ["  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current", "                                 Dload  Upload   Total   Spent    Left  Speed", "", "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (6) Could not resolve host: GET; No address associated with hostname", "", "100  1041  100  1041    0     0   708k      0 --:--:-- --:--:-- --:--:--  708k"], "stdout": "RW\nRW", "stdout_lines": ["RW", "RW"]}

TASK [Test Passed] *****************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:282
ok: [localhost] => {"ansible_facts": {"flag": "Test Passed"}, "changed": false}

TASK [include_tasks] ***************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-cleanup.yml:2
included: /root/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml for localhost

TASK [Deploy application files] ****************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/delete_deploy.yml:2
changed: [localhost -> None] => (item=percona.yaml) => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f percona.yaml -n default", "delta": "0:00:03.358746", "end": "2018-05-24 18:36:32.907134", "item": "percona.yaml", "rc": 0, "start": "2018-05-24 18:36:29.548388", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "deployment \"percona\" deleted\npersistentvolumeclaim \"demo-vol1-claim\" deleted\nservice \"percona-mysql\" deleted", "stdout_lines": ["deployment \"percona\" deleted", "persistentvolumeclaim \"demo-vol1-claim\" deleted", "service \"percona-mysql\" deleted"]}

TASK [include_tasks] ***************************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-cleanup.yml:7
included: /root/openebs/e2e/ansible/playbooks/utils/delete_deploy_check.yml for localhost

TASK [Check pod status] ************************************************************************************************************************************************
task path: /root/openebs/e2e/ansible/playbooks/utils/delete_deploy_check.yml:2
changed: [localhost -> None] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl get pods -n default | grep percona", "delta": "0:00:00.279898", "end": "2018-05-24 18:36:33.710904", "rc": 0, "start": "2018-05-24 18:36:33.431006", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "percona-2897667160-zp7p5                                        0/1       Terminating   4          8m", "stdout_lines": ["percona-2897667160-zp7p5                                        0/1       Terminating   4          8m"]}

TASK [Remove the percona liveness check config map] ********************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-cleanup.yml:12
changed: [localhost -> None] => {"changed": true, "cmd": "source ~/.profile; kubectl delete cm sqltest -n default", "delta": "0:00:00.928525", "end": "2018-05-24 18:36:35.227874", "failed_when_result": false, "rc": 0, "start": "2018-05-24 18:36:34.299349", "stderr": "/bin/bash: /root/.profile: No such file or directory", "stderr_lines": ["/bin/bash: /root/.profile: No such file or directory"], "stdout": "configmap \"sqltest\" deleted", "stdout_lines": ["configmap \"sqltest\" deleted"]}

TASK [Remove test artifacts] *******************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure-cleanup.yml:26
changed: [localhost -> None] => (item=percona.yaml) => {"changed": true, "item": "percona.yaml", "path": "/root/percona.yaml", "state": "absent"}
changed: [localhost -> None] => (item=sql-test.sh) => {"changed": true, "item": "sql-test.sh", "path": "/root/sql-test.sh", "state": "absent"}
changed: [localhost -> None] => (item=storage-pool-path_creation.yml) => {"changed": true, "item": "storage-pool-path_creation.yml", "path": "/root/storage-pool-path_creation.yml", "state": "absent"}
changed: [localhost -> None] => (item=storage-class.yaml) => {"changed": true, "item": "storage-class.yaml", "path": "/root/storage-class.yaml", "state": "absent"}

TASK [Test Cleanup Passed] *********************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:297
ok: [localhost] => {"ansible_facts": {"cflag": "Cleanup Passed"}, "changed": false}

TASK [Send slack notification] *****************************************************************************************************************************************
task path: /home/sathya/openebs/e2e/ansible/playbooks/resiliency/test-disk-failure/test-disk-failure.yml:314
skipping: [localhost] => {"changed": false, "skip_reason": "Conditional result was False"}
META: ran handlers
META: ran handlers

PLAY RECAP *************************************************************************************************************************************************************
localhost                  : ok=44   changed=33   unreachable=0    failed=0

```
